nixops_encrypted_links/lib.py

```diff
@@ -180,6 +167,17 @@ def do_machine(m: nixops.backends.MachineState) -> None:
     def emit_resource(r: nixops.resources.ResourceState) -> None:
         config = attrs_per_resource[r.name]
         if is_machine(r):
+            # Sort the hosts by its canonical host names.
+            sorted_hosts = sorted(
+                hosts[r.name].items(), key=lambda item: item[1][0]
+            )
+
+            # Just to remember the format:
+            #   ip_address canonical_hostname [aliases...]
+            extra_hosts = {
+                f"{ip}": names for ip, names in sorted_hosts
+            }
+
```

Comment:
170 - 180 and 201 lines required to get the new encrypted-links hosts added to the target machines.